core/src/main/scala/org/apache/spark/status/AppStatusListener.scala

```diff
@@ -228,9 +230,12 @@ private[spark] class AppStatusListener(
 
     // Implicitly blacklist every available executor for the stage associated with this node
     Option(liveStages.get((event.stageId, event.stageAttemptId))).foreach { stage =>
-      liveExecutors.values.filter(_.host == event.hostId).foreach { exec =>
-        setStageBlackListStatus(stage, exec.executorId, now)
+      val executorIds = liveExecutors.values.filter(_.host == event.hostId).map(_.executorId)
+      executorIds.foreach { executorId =>
+        setStageBlackListStatus(stage, executorId, now)
       }
+      stage.blackListedExecutors ++= executorIds
+      maybeUpdate(stage, now)
```

Comment:
if `setStageBlacklistStatus` took a Seq or varags for multiple executorIds, you could push the these other updates down into it.

Reply:
Good idea.